### PR TITLE
Generate OpenCL C feature dictionary

### PR DESCRIPTION
**Pull request by kpet**
Features are stored in a text file for now. Ultimately, we probably want to use the XML registry for this.

Generation script taken from #1174 with a few modifications.

Contributes to #1166.

Change-Id: Ie2c14148d75457030aa1a97cf601daba2c007397

### CODE REVIEW

File: scripts/gen_c_feature_dictionary.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                        default='',
+                        help='File with OpenCL C features to generate, one per line')
+    parser.add_argument('-o', action='store', default='',
+                        help='Output file in which to store the feature dictionary. stdout is used if no file is provided.')
+
+    args = parser.parse_args()
+
+    features = []
+    if len(args.features) > 0:
+        print('Generating feature dictionaries from: ' + args.features + ' ...\n\n')
```
Comment (bashbaug) [inline scripts/gen_c_feature_dictionary.py:24]: Nitpick: I'd prefer not to have additional newlines here, to be consistent with the other dictionary generation scripts.

```suggestion
        print('Generating feature dictionaries from: ' + args.features)
```

**Review by bashbaug — COMMENTED** [review]
This is a nice improvement!

Once this goes in I might look at generalizing it for other uses in the OpenCL C spec.  For example, it might be useful to have dictionaries for long built-in function names, to fix https://github.com/KhronosGroup/OpenCL-Docs/issues/1214.

File: c/features.txt [hunk 2]
```diff
@@ -0,0 +1,21 @@
+__opencl_c_feature_name
```
[the remark this answers is not in the document]
Comment (kpet) [reply c/features.txt]: Well spotted! Agree (2) is the way to go. This has the added benefit of removing what is not really a feature from the list of features, making the transition to the XML easier. I've now done that, PTAL.

**Review by bashbaug — APPROVED** [review]
Thanks!

**Pull request merged**